File: client/lib/models/subject_marks.dart
class SubjectMarks {
  final String subjectName;
  final String maxMarksTheory;
  final String maxMarksInternal;
  final String maxMarksTotal;
  final String minMarksTheory;
  final String minMarksInternal;
  final String obtainedTheory;
  final String obtainedInternal;
  final String totalMarks;
  final String remarks;

  SubjectMarks({
    required this.subjectName,
    required this.maxMarksTheory,
    required this.maxMarksInternal,
    required this.maxMarksTotal,
    required this.minMarksTheory,
    required this.minMarksInternal,
    required this.obtainedTheory,
    required this.obtainedInternal,
    required this.totalMarks,
    this.remarks = '',
  });

  /// Uses short keys to minimize QR data payload size.
  Map<String, dynamic> toMap() {
    return {
      'n': subjectName,
      'a': maxMarksTheory,
      'b': maxMarksInternal,
      'c': maxMarksTotal,
      'd': minMarksTheory,
      'e': minMarksInternal,
      'f': obtainedTheory,
      'g': obtainedInternal,
      't': totalMarks,
      'r': remarks,
    };
  }

  factory SubjectMarks.fromMap(Map<String, dynamic> map) {
    return SubjectMarks(
      subjectName: map['n'] ?? '',
      maxMarksTheory: map['a'] ?? '',
      maxMarksInternal: map['b'] ?? '',
      maxMarksTotal: map['c'] ?? '',
      minMarksTheory: map['d'] ?? '',
      minMarksInternal: map['e'] ?? '',
      obtainedTheory: map['f'] ?? '',
      obtainedInternal: map['g'] ?? '',
      totalMarks: map['t'] ?? '',
      remarks: map['r'] ?? '',
    );
  }
}

File: Domain/Entities/SubjectMarks.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Domain.Entities;

public class SubjectMarks
{
    [Key]
    public int Id { get; set; }

    public string StudentResultRollNumber { get; set; } = string.Empty; // Foreign Key

    [JsonPropertyName("n")]
    public string SubjectName { get; set; } = string.Empty;

    [JsonPropertyName("a")]
    public string MaxMarksTheory { get; set; } = string.Empty;

    [JsonPropertyName("b")]
    public string MaxMarksInternal { get; set; } = string.Empty;

    [JsonPropertyName("c")]
    public string MaxMarksTotal { get; set; } = string.Empty;

    [JsonPropertyName("d")]
    public string MinMarksTheory { get; set; } = string.Empty;

    [JsonPropertyName("e")]
    public string MinMarksInternal { get; set; } = string.Empty;

    [JsonPropertyName("f")]
    public string ObtainedTheory { get; set; } = string.Empty;

    [JsonPropertyName("g")]
    public string ObtainedInternal { get; set; } = string.Empty;

    [JsonPropertyName("t")]
    public string TotalMarks { get; set; } = string.Empty;

    [JsonPropertyName("r")]
    public string Remarks { get; set; } = string.Empty;
}